### ISSUE TITLE
Pivoting from SQL or Supporting Multiple Tables in Native Questions

### ISSUE THREAD

**Issue by tyliechty** [metabase/metabase#2324]
If this is a bug report, please fill in the blanks:

*  I am using the Chrome browser.
*  My computer's OS is Windows 10.
*  I'm running Metabase from a Local Docker Image on Openstack.
*  My database is BigQuery.
*  My Metabase version is v0.16.0.

Our BigQuery data comes in as a new table everyday. We can not select across multiple tables without using SQL directly, and with SQL can not Pivot the data.

**Comment by agilliland** [comment]
Hmm, that definitely sounds like an interesting use case.  So you have a set of tables such as `events_20160405`, `events_20160406`, etc, and you are looking for a way to run a query against multiple days worth of data and then pivot it?

**Comment by tyliechty** [comment]
That is correct, in the SQL the line that handles multiple tables/date ranges looks like this:

FROM (TABLE_DATE_RANGE([foo.bar_], TIMESTAMP('2016-01-01'), current_timestamp()))

**Comment by agilliland** [comment]
Interesting.  Admittedly I'm not super familiar with BigQuery, but it sounds like this is a general feature for accessing a partitioned table in BQ and ideally Metabase would support this so you don't have to think about it.  Here's the relevant bit from the BQ docs ...

> The term table wildcard function refers to a special type of function unique to BigQuery. These functions are used in the FROM clause to match a collection of table names using one of several types of filters. For example, the TABLE_DATE_RANGE function can be used to query only a specific set of daily tables. For more information on these functions, see Table wildcard functions.

I'm assuming that only certain tables are partitioned this way correct?  So there would need to be some way to mark them as such so that they could be handled properly.

**Comment by zacman85** [comment]
While using custom date ranges may be important to more than a few folks, we have had decent success using Bigquery views to encapsulate rolling time ranges. For instance, most of our per-day partitioned tables have "trailing_quarter", "trailing_twomonths", "trailing_month", "trailing_twoweeks", and "trailing_week" views. These can actually be bound into Metabase, but I haven't gotten it to completely work yet. Will dig in more and possibly find/add an issue on the subject.

**Comment by camsaul** [comment]
This will need a special UI widget for selecting a table date range

**Comment by edouardKaiser** [comment]
We are very interested in that feature too.

We were really excited about using metabase for our dashboard solution on top of BigQuery, Unfortunately the fact that we can't query across a table range, was a no-go for us.

**Comment by camsaul** [comment]
@edouardKaiser while our query builder doesn't support `TABLE_DATE_RANGE` you can always switch to SQL, write whatever queries you want, visualize the results, and add those to your dashboards.

Our query builder supports a ton of use cases but lots of people still use SQL when they need to do more advanced things.

**Comment by florentcailly** [comment]
It would definitely be a game changer !

**Comment by kfleming-target** [comment]
This would be great for us as well - we're using views at the moment but BigQuery doesn't support caching for views which makes it much more expensive to run as the data grows.
